storage/src/main/scala/coop/rchain/storage/Key.scala

```diff
@@ -112,7 +119,6 @@ class Key(keyIn: String) extends TermTree with Ordered[Key] {
     bindings
   }
 
-  // sort by: arity, character, digit, other
   def compare(key: Key): Int = {
     if (term == key.term) { return 0 }
 
```

Comment:
```scala
var i = 0, result = 0
do {
  // instead of return, set result
  ...
  i += 1
} while (i < length && result == 0)
if (result != 0) {
  result
} else {
  assert(term.length != key.term.length)
  if (term.length > key.term.length) { 1 }
  else { -1 }
}
```

Reply:
Could also use recursion